Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -436,29 +436,38 @@ bool SVertexer::checkV0(const TrackCand& seedP, const TrackCand& seedN, int iP,
 }
 
 //__________________________________________________________________
-int SVertexer::checkCascades(float rv0, std::array<float, 3> pV0, float p2V0, int avoidTrackID, int posneg, int ithread)
+int SVertexer::checkCascades(float rv0, std::array<float, 3> pV0, float p2V0, int avoidTrackID, int posneg, VBracket v0vlist, int ithread)
 {
+
   // check last added V0 for belonging to cascade
   auto& fitterCasc = mFitterCasc[ithread];
   const auto& v0 = mV0sTmp[ithread].back();
   auto& tracks = mTracksPool[posneg];
-  const auto& pv = mPVertices[v0.getVertexID()];
+
   int nCascIni = mCascadesTmp[ithread].size();
   // start from the 1st track compatible with V0's primary vertex
-  int firstTr = mVtxFirstTrack[posneg][v0.getVertexID()], nTr = tracks.size();
+  int firstIdx = v0vlist.getMin() - mSVParams->marginBachPV > 0 ? v0vlist.getMin() - mSVParams->marginBachPV : 0;
```

Comment:
Could you clarify why this marginBachPV is needed? By taking the `firstIdx = v0vlist.getMin()` you are already picking the 1st track compatible with the earliest vertex compatible with v0. Why do you need to go to earlier tracks?

Reply:
Because bachelors are sorted by minimum vtxID.  If the array starts here: `mVtxFirstTrack[posneg][v0.getMin()]`  you miss all the bachelors having a minimum vtx id < v0.getMin(). Then when you take the overlap between the two brackets is already too late, and you are getting the overlap only for bachelors w/ min vertexID >= v0.getMin(). Do you see my point?

Reply:
OK, I think I understood, but then any fixed offset will be either insufficient starting from some interaction rate (vertices per millisecond) or will kill the advantage of using sorted indices. Would not it better in this case to prepare in advance an extra index vector for bachelors with inverse sorting (by the highest vertex ID compatible with this track), so that all the tracks below `mVtxFirstTrack[posneg][ invBachIndex[v0vlist.getMin] ]` have no chance to be compatible with this vertex and those after?

Reply:
Yes, you are right! I am working on it